### ISSUE TITLE
Events disappearing after a while

### ISSUE THREAD

**Issue by andrevmatos** [raiden-network/raiden#808]
## Problem Definition

Inducing a channel event to be fired (for example, by depositing to the channel) causes the event to appear on channel events API. But, after a small time (some seconds), fetching this same event, without `from_block` parameter, returns empty list. Looks like Raiden is _forgeting_ past events.

## Example

Deposit 3 tokens to the channel:
```
$ curl 'http://localhost:5001/api/1/channels/0xd54195db14a12b79ae33b23fdc75df1ebe79d277' -X PATCH -H 'Content-Type: application/json' --data-binary '{"balance":3}'
{
  "balance": 10, 
  "channel_address": "0xd54195db14a12b79ae33b23fdc75df1ebe79d277", 
  "partner_address": "0x000d91cf263a11f9bfcee3752e5b03fc1196ce98", 
  "settle_timeout": 600, 
  "state": "opened", 
  "token_address": "0x9aba529db3ff2d8409a1da4c9eb148879b046700"
}
```
Check events:
```
$ curl http://localhost:5001/api/1/events/channels/0xd54195db14a12b79ae33b23fdc75df1ebe79d277
[{"block_number": 0, "participant": "b0d5eeaeb7b084a2ba92c24b79f4f2d079a3062b", "event_type": "ChannelNewBalance", "balance": 24, "token_address": "9aba529db3ff2d8409a1da4c9eb148879b046700"}]
```
Check events again after 30s:
```
$ curl http://localhost:5001/api/1/events/channels/0xd54195db14a12b79ae33b23fdc75df1ebe79d277
[]
```

**Another** issue can be seen in this example: as seen in the `PATCH` request, channel `balance` isn't being updated, and following requests to `/api/1/channels` also return old balance.

## Solution

Make sure the events are being persisted and remembered, at least during session, but preferably, across sessions. Also, guarantee channel balances are being correctly updated.

## System Description
Master 1d6633dbb9ef343675fc52950f2aee85d97504a9 , from today.

**Comment by ulope** [comment]
I've too seen this intermittently on the testnet but haven't found (and also not really looked yet) a repeatable way to reproduce.

**Comment by LefterisJP** [comment]
@andrevmatos The secondary problem you reported with this issue:

> Another issue can be seen in this example: as seen in the PATCH request, channel balance isn't being updated, and following requests to /api/1/channels also return old balance.

 will be addressed by #824 

In the future please open different github issues for different .. ehm ... issues :P

**Comment by andrevmatos** [comment]
Usually, I do, but as the example (curl calls, etc) were the same, and showed clearly both issues, and they were related (see [my comment](https://github.com/raiden-network/raiden/pull/824#issuecomment-318045456)), so I did it, but will try to opening different issues even with mostly the same data. Thank you.

**Comment by andrevmatos** [comment]
`from_block` was defaulting to `latest`, so it only showed events from the latest block (a few seconds). When a new one arrived, it filtered out the previously seen events. Also, lots of events were coming without block_number. #836 should fix this too.

@LefterisJP , despite that, in a test I did with the correct block_numbers coming on events API, I tested making a deposit, `ChannelNewBalance` event poped up correctly on webUI, but in Raiden, `EthNodeCommunicationError: After 60 seconds the deposit event was not seen by the ethereum node.` still happens. Any idea why seems like Raiden isn't being able to listen to [this] event[s]?
